packages/nimble-components/src/rich-text/viewer/tests/rich-text-viewer.spec.ts

```diff
@@ -461,37 +545,31 @@ describe('RichTextViewer', () => {
 
         it('should return updated href when mention configuration element pattern get updated', async () => {
             element.markdown = '<user:1>';
-            await appendUserMentionConfiguration(
-                element,
-                ['user:1'],
-                ['username1']
-            );
-            await waitForUpdatesAsync();
+            await appendUserMentionConfiguration(element, [
+                { key: 'user:1', displayName: 'username1' }
+            ]);
             const renderedUserMention = element.lastElementChild as RichTextMentionUsers;
             expect(element.getMentionedHrefs()).toEqual(['user:1']);
             renderedUserMention.pattern = 'invalid';
+            await waitForUpdatesAsync();
             expect(element.getMentionedHrefs()).toEqual([]);
         });
 
         it('should return updated href when mention configuration element added dynamically', async () => {
             element.markdown = '<user:1>';
-            await appendUserMentionConfiguration(
-                element,
-                ['user:1'],
-                ['username1']
-            );
+            await appendUserMentionConfiguration(element, [
+                { key: 'user:1', displayName: 'username1' }
+            ]);
             await waitForUpdatesAsync();
```

Comment:
Looks like `appendUserMentionConfiguration` calls `waitForUpdatesAsync` before returning. Review all the call sites and remove any redundant calls to `waitForUpdatesAsync` that immediately follow. (If for some reason a second `waitForUpdatesAsync` call is actually necessary, please add a comment explaining that.)

Reply:
Thanks for pointing it out. I have removed `waitForUpdateAsync` after `appendUserMentionConfiguration` calls in every place we use in this file.